server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -136,6 +136,8 @@ private static Version parseUnchecked(String version) {
     public static final IndexVersion LOGSB_OPTIONAL_SORTING_ON_HOST_NAME_BACKPORT = def(8_525_0_00, parseUnchecked("9.12.1"));
     public static final IndexVersion USE_SYNTHETIC_SOURCE_FOR_RECOVERY_BY_DEFAULT_BACKPORT = def(8_526_0_00, parseUnchecked("9.12.1"));
     public static final IndexVersion SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_KEYWORD_BACKPORT_8_X = def(8_527_0_00, Version.LUCENE_9_12_1);
+    public static final IndexVersion SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_NUMBER_BACKPORT_8_X = def(8_528_0_00, Version.LUCENE_9_12_1);
+    public static final IndexVersion SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_BOOLEAN_BACKPORT_8_X = def(8_529_0_00, Version.LUCENE_9_12_1);
```

Comment:
I think this version doesn't exist in the 8.x branch?

Thinking more about this, given the 8.19.0 has not been released yet. I think we reuse the `SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_KEYWORD_BACKPORT_8_X` index version. Also in 8.x branch, however I see that `SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_NUMBER` was added. So In think here we can just use `SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_NUMBER` in places where `SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_BOOLEAN_BACKPORT_8_X ` is used?

Reply:
When I've been backporting the PRs adding offset encoding, I've been creating new index versions for each backport. The index version `SYNTHETIC_SOURCE_STORE_ARRAYS_NATIVELY_BOOLEAN_BACKPORT_8_X` (`8_529`) is still pending in backport PR #125596.

Instead of adding a new version, should I update #125596 to instead use `8_528` for the boolean field mapper, since there hasn't been an 8.x release yet? 

If so, should I also use `8_528` for future offset encoding backports (unsigned long and scaled float)?

Then I would update this PR to use `8_528` for both numeric and boolean fields.

Reply:
OK, I opened #125610 to use `8_527` for all the 8.x offset encoding implementations. I also updated the version checks here to always just look at `8_527`